src/main/java/ai/shreds/adapter/AdapterKafkaProducerException.java

```diff
@@ -0,0 +1,26 @@
+package ai.shreds.adapter.dto;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+/**
+ * Data Transfer Object (DTO) for encapsulating error responses in the adapter layer.
+ */
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+public class AdapterSharedErrorResponse {
```

Comment:
This must be AdapterKafkaProducerException class not AdapterSharedErrorResponse. remove all the code and regenerate the code for the AdapterKafkaProducerException